.github/workflows/type-check-backend.yml

```diff
@@ -0,0 +1,28 @@
+name: Type Check Backend
+on:
+  pull_request:
+
+jobs:
+  checking:
+    name: Type Checking Backend
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
```

Comment:
Because we install dependencies in test-backend as well, I wonder if there's a way to cache that bit to reduce time and network utilisation, but that could be one for future improvements

Reply:
Yeah I think it's a trade off situation. We definitely could cache it but that would probably entail us linking the two actions together somehow or even combining them into one. We could definitely do that, but it's possible not being able to run the two actions in parallel could end up being slower.